Mathlib/Data/ZMod/Basic.lean

```diff
@@ -1293,3 +1293,13 @@ def Nat.residueClassesEquiv (N : ℕ) [NeZero N] : ℕ ≃ ZMod N × ℕ where
         cast_id', id_eq, zero_add]
     · simp only [add_comm p.1.val, mul_add_div (NeZero.pos _),
         (Nat.div_eq_zero_iff).2 <| .inr p.1.val_lt, add_zero]
+
+-- there is a faster proof with Module.toAddMonoidEnd
+instance ZMod.instSubsingletonModule (n : ℕ) (M : Type*) [AddCommMonoid M] :
+    Subsingleton (Module (ZMod n) M) := by
+  obtain _ | n := n
+  · exact inferInstanceAs (Subsingleton (Module ℤ M))
+  refine ⟨fun m1 m2 ↦ Module.ext' _ _ fun r m ↦ ?_⟩
+  obtain ⟨r, rfl⟩ := ZMod.natCast_zmod_surjective r
+  conv_lhs => exact letI := m1; Nat.cast_smul_eq_nsmul _ r m
+  conv_rhs => exact Nat.cast_smul_eq_nsmul _ r m
```

Comment:
```suggestion
  rw [(letI := m1; Nat.cast_smul_eq_nsmul _ r m), Nat.cast_smul_eq_nsmul _ r m]
```
The `()` are optional, but they help readability IMO. In any case in think `conv_lhs => exact` should be avoided as it’s a rather convoluted way of spelling an `erw`... (thankfully, in this case this is a plain `rw`).